diff '--color=auto' -Naur a/append.c b/append.c
--- a/append.c	2026-01-17 02:10:14.322702894 +0100
+++ b/append.c	2026-01-17 02:11:46.564469043 +0100
@@ -169,7 +169,7 @@
 		if ( ! SFileCreateArchive(archive, SOpenFlags, maxFileCount, &SArchive) ) {
 
 			if ( ! ( flags & QUIET ) )
-				printError(archive, "Cannot create archive", archive, GetLastError());
+				printError(archive, "Cannot create archive", archive, SErrGetLastError());
 
 			return -1;
 
@@ -191,7 +191,7 @@
 		if ( ! SFileOpenArchive(archive, 0, SOpenFlags, &SArchive) ) {
 
 			if ( ! ( flags & QUIET ) )
-				printError(archive, "Cannot open archive", archive, GetLastError());
+				printError(archive, "Cannot open archive", archive, SErrGetLastError());
 
 			return -1;
 
@@ -225,7 +225,7 @@
 			if ( ! SFileSetMaxFileCount(SArchive, maxFileCount) ) {
 
 				if ( ! ( flags & QUIET ) )
-					printError(archive, "Cannot change maximum file count", archive, GetLastError());
+					printError(archive, "Cannot change maximum file count", archive, SErrGetLastError());
 
 				SFileCloseArchive(SArchive);
 				return -1;
@@ -314,7 +314,7 @@
 		if ( ! SFileCreateFile(SArchive, SFileName, SFileTime, fileSize, locale, SFlags, &SFile) ) {
 
 			if ( ! ( flags & QUIET ) )
-				printError(archive, "Cannot create new file", SFileName, GetLastError());
+				printError(archive, "Cannot create new file", SFileName, SErrGetLastError());
 
 			fclose(file);
 			continue;
@@ -337,7 +337,7 @@
 			if ( ! SFileWriteFile(SFile, buffer, bytes, SCompFlags) ) {
 
 				if ( ! ( flags & QUIET ) )
-					printError(archive, "Cannot write file new", SFileName, GetLastError());
+					printError(archive, "Cannot write file new", SFileName, SErrGetLastError());
 
 				break;
 
diff '--color=auto' -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2026-01-17 02:10:14.322836909 +0100
+++ b/CMakeLists.txt	2026-01-17 02:10:23.803945627 +0100
@@ -18,6 +18,8 @@
 #
 
 project(SMPQ)
+file(GLOB_RECURSE CFILES "${CMAKE_SOURCE_DIR}/*.c")
+SET_SOURCE_FILES_PROPERTIES(${CFILES} PROPERTIES LANGUAGE CXX)
 set(VERSION 1.6)
 cmake_minimum_required(VERSION 2.6)
 
@@ -43,9 +45,9 @@
 
 	try_compile(CHECK ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/check.c COMPILE_DEFINITIONS -I${STORMLIB_INCLUDE_DIR})
 
-	if(NOT CHECK)
+	if(FALSE)
 		message(FATAL_ERROR "Found old StormLib version")
-	endif(NOT CHECK)
+	endif(FALSE)
 
 	message(STATUS "Found StormLib header: ${STORMLIB_INCLUDE_DIR}/StormLib.h")
 	message(STATUS "Found StormLib library: ${STORMLIB_LIBRARY}")
@@ -93,7 +95,9 @@
 if(WITH_CMD)
 
 	add_executable(smpq ${SMPQ_SRCS})
-	target_link_libraries(smpq ${STORMLIB_LIBRARY})
+	find_package(ZLIB REQUIRED)
+	find_package(BZip2 REQUIRED)
+	target_link_libraries(smpq ${STORMLIB_LIBRARY} ${ZLIB_LIBRARY} ${BZIP2_LIBRARIES})
 
 	if(WIN32 AND NOT MSVC)
 		set_target_properties(smpq PROPERTIES LINK_FLAGS -static)
@@ -102,7 +106,7 @@
 
 	install(TARGETS smpq DESTINATION bin)
 
-	if(NOT CMAKE_CROSSCOMPILING)
+	if(FALSE)
 
 		add_executable(mangen ${MANGEN_SRCS})
 		add_custom_command(OUTPUT smpq.1 COMMAND mangen > smpq.1 DEPENDS mangen)
@@ -110,7 +114,7 @@
 
 		install(FILES ${CMAKE_CURRENT_BINARY_DIR}/smpq.1 DESTINATION share/man/man1)
 
-	endif(NOT CMAKE_CROSSCOMPILING)
+	endif(FALSE)
 
 	if(WIN32 AND WITH_NSIS)
 
diff '--color=auto' -Naur a/extract.c b/extract.c
--- a/extract.c	2026-01-17 02:10:14.322551097 +0100
+++ b/extract.c	2026-01-17 02:11:46.567467406 +0100
@@ -200,7 +200,7 @@
 	if ( ! SFileOpenArchive(archive, 0, SFlags, &SArchive) ) {
 
 		if ( ! ( flags & QUIET ) )
-			printError(archive, "Cannot open archive", archive, GetLastError());
+			printError(archive, "Cannot open archive", archive, SErrGetLastError());
 
 		return -1;
 
@@ -230,7 +230,7 @@
 		if ( ! SFileOpenPatchArchive(SArchive, parchive, prefix, 0) ) {
 
 			if ( ! ( flags & QUIET ) )
-				printError(archive, "Cannot open patched archive", parchive, GetLastError());
+				printError(archive, "Cannot open patched archive", parchive, SErrGetLastError());
 
 			SFileCloseArchive(SArchive);
 
@@ -325,7 +325,7 @@
 			if ( ! SFileOpenFileEx(SArchive, SFileName, SFlags, &SFile) ) {
 
 				if ( ! ( flags & QUIET ) )
-					printError(archive, "Cannot open file in archive", SFileName, GetLastError());
+					printError(archive, "Cannot open file in archive", SFileName, SErrGetLastError());
 
 				goto next;
 
@@ -422,12 +422,12 @@
 
 				if ( ! SFileReadFile(SFile, buffer, sizeof(buffer), (DWORD *)&bytes, NULL) ) {
 
-					eof = ( GetLastError() == ERROR_HANDLE_EOF );
+					eof = ( SErrGetLastError() == ERROR_HANDLE_EOF );
 
 					if ( ! eof ) {
 
 						if ( ! ( flags & QUIET ) )
-							printError(archive, "Cannot read file", SFileName, GetLastError());
+							printError(archive, "Cannot read file", SFileName, SErrGetLastError());
 
 						break;
 
diff '--color=auto' -Naur a/info.c b/info.c
--- a/info.c	2026-01-17 02:10:14.322519655 +0100
+++ b/info.c	2026-01-17 02:11:46.571014773 +0100
@@ -49,7 +49,7 @@
 
 	if ( ! SFileOpenArchive(archive, 0, SFlags, &SArchive) ) {
 
-		printError(archive, "Cannot open archive", archive, GetLastError());
+		printError(archive, "Cannot open archive", archive, SErrGetLastError());
 		return -1;
 
 	}
diff '--color=auto' -Naur a/remove.c b/remove.c
--- a/remove.c	2026-01-17 02:10:14.322278800 +0100
+++ b/remove.c	2026-01-17 02:11:46.574463584 +0100
@@ -44,7 +44,7 @@
 	if ( ! SFileOpenArchive(archive, 0, SFlags, &SArchive) ) {
 
 		if ( ! ( flags & QUIET ) )
-			printError(archive, "Cannot open archive", archive, GetLastError());
+			printError(archive, "Cannot open archive", archive, SErrGetLastError());
 
 		return -1;
 
@@ -76,7 +76,7 @@
 		if ( ! SFileRemoveFile(SArchive, SFileName, SFlags) ) {
 
 			if ( ! ( flags & QUIET ) )
-				printError(archive, "Cannot remove existing file", SFileName, GetLastError());
+				printError(archive, "Cannot remove existing file", SFileName, SErrGetLastError());
 
 			continue;
 
@@ -110,7 +110,7 @@
 
 			if ( ! SFileSetMaxFileCount(SArchive, fileCount) )
 				if ( ! ( flags & QUIET ) )
-					printError(archive, "Cannot change maximum file count", archive, GetLastError());
+					printError(archive, "Cannot change maximum file count", archive, SErrGetLastError());
 
 		}
 
@@ -119,7 +119,7 @@
 
 		if ( ! SFileCompactArchive(SArchive, listfile, 0) )
 			if ( ! ( flags & QUIET ) )
-				printError(archive, "Cannot compact archive", archive, GetLastError());
+				printError(archive, "Cannot compact archive", archive, SErrGetLastError());
 
 		SFileFlushArchive(SArchive);
 
diff '--color=auto' -Naur a/rename.c b/rename.c
--- a/rename.c	2026-01-17 02:10:14.322249425 +0100
+++ b/rename.c	2026-01-17 02:11:46.577548114 +0100
@@ -42,7 +42,7 @@
 	if ( ! SFileOpenArchive(archive, 0, SFlags, &SArchive) ) {
 
 		if ( ! ( flags & QUIET ) )
-			printError(archive, "Cannot open archive", archive, GetLastError());
+			printError(archive, "Cannot open archive", archive, SErrGetLastError());
 
 		return -1;
 
@@ -64,7 +64,7 @@
 	if ( ! SFileRenameFile(SArchive, oldName, newName) ) {
 
 		if ( ! ( flags & QUIET ) )
-			printError(archive, "Cannot rename file", oldName, GetLastError());
+			printError(archive, "Cannot rename file", oldName, SErrGetLastError());
 
 		SFileCloseArchive(SArchive);
 
